include/range/v3/detail/config.hpp

```diff
@@ -291,6 +291,42 @@
 #define RANGES_GCC_BROKEN_CUSTPOINT
 #endif
 
+#define RANGES_CXX_INLINE_VARIABLES_11 0
+#define RANGES_CXX_INLINE_VARIABLES_14 0
+#define RANGES_CXX_INLINE_VARIABLES_17 201606
+#ifndef RANGES_CXX_INLINE_VARIABLES
+
+#ifdef __cpp_inline_variables // TODO: fix this if SD-6 picks another name
+#define RANGES_CXX_INLINE_VARIABLES __cpp_inline_variables
+#elif defined(__clang__) && \
+    (__clang__major__ > 3 || __clang_major__ == 3 && __clang_minor__ == 9) && \
+    __cplusplus > 201402L
+// TODO: remove once clang defines __cpp_inline_variables (or equivalent)
+#define RANGES_CXX_INLINE_VARIABLES RANGES_CXX_INLINE_VARIABLES_17
+#else
+#define RANGES_CXX_INLINE_VARIABLES RANGES_CXX_FEATURE(INLINE_VARIABLES)
+#endif  // __cpp_inline_variables
+
+#endif  // RANGES_CXX_INLINE_VARIABLES
+
+#if RANGES_CXX_INLINE_VARIABLES < RANGES_CXX_INLINE_VARIABLES_17
+#define RANGES_INLINE_VARIABLE_(type, name, inline_)      \
+    inline_ namespace                                     \
+    {                                                     \
+        constexpr auto& name = static_const<type>::value; \
+    }                                                     
+
+#else  // RANGES_CXX_INLINE_VARIABLES >= RANGES_CXX_INLINE_VARIABLES_17
+#define RANGES_INLINE_VARIABLE_(type, name, inline_) \
+    namespace function_objects                       \
```

Comment:
And intentionally not used here. The point of `inline_` is to insert `RANGES_GCC_BROKEN_CUSTPOINT` into the definition of `RANGES_INLINE_VARIABLE_` only when inline variables are not supported. (I'm not sure why @gnzlbg passes it in through a parameter instead of directly putting `RANGES_GCC_BROKEN_CUSTPOINT` in the definition of `RANGES_INLINE_VARIABLE`, but it does not seem worth the trouble to change.)